tools/guided-fuzzing/src/GuidedExecutionPolicy.cpp

```diff
@@ -17,7 +22,22 @@ bool GuidedExecutionPolicy::should_queue_path(const Context& ctx) {
     combined.insert(i);
   }
 
-  return solver_->check(combined) != SolverResult::UNSAT;
+  if (mutator->solver->check(combined) == SolverResult::SAT) {
+    return true;
+  }
+
+  // If the path that we're checking isn't SAT given the assumptions, we want
+  // to see if it's SAT in general without our additional assumption so that
+  // we can create a testcase out of it. The key here is that the
+  // `NeverQueuePolicy` will turn our `ctx_copy` into a testcase after it is
+  // done running (assuming it is SAT)
+  auto store = NopStore();
+  auto policy = NeverQueuePolicy(cases, mutator);
+  auto ctx_copy = ctx;
+  Interpreter interp(&ctx_copy, &policy, &store, nullptr, mutator->solver);
+  interp.execute();
```

Comment:
I don't think this will work properly. Maybe you wanted this instead? It seems to do what you want.
```suggestion
  // If the path that we're checking isn't SAT given the assumptions, we want
  // to see if it's SAT in general without our additional assumption so that
  // we can create a testcase out of it. 
  auto policy = NeverQueuePolicy(cases, mutator);
  policy.on_path_complete(ctx, ExitStatus::Dead);
```

Reply:
The reason I'm creating a new interpreter is because I want to see if the `Context` being passed in is `SAT` when executed (without additional assertions).

Reply:
After some offline discussion I see your point. Also created #401 to fix some issues related to this

Reply:
Creating a new interpreter would continue executing along the path. I'm going to try and draw this out with some diagrams:

So suppose the main path through the program looks sorta like this (horizontal for compactness)
```
____________________________________________________________
```
and then we have branches that go off but aren't the same path as taken by the test case
```
____________________________________________________________
    \                           \                       \
```

If we don't create a new interpreter we're just creating test cases for those branches that go off.

However, if we create a new interpreter and keep executing we're doing something like this:
```
____________________________________________________________
      \                   \                         \____________________
       \                   \_________                    \            \
        \                       \
         \__________
             \     \
```
That is, we don't just follow the main test case, we also follow every branch off of it until the next branch.

What I think you want is the first one. Continuing executing branches that aren't the main one doesn't tell if you anything more about whether it's currently SAT than just running the solver over the current path condition (`ctx.assertions`). The only exception to this is if you have a failure via `logger->log_failure` in which case you also need to include the extra assertion that gets passed to log_failure.